src/FTPWindow.cpp

```diff
@@ -844,17 +842,14 @@ void TFTPWindow::CopyUrl()
 		= ((BStringField *)row->GetField(CLM_INTERNAL_NAME))->String();
 
 	BString url;
-	if (fCurrentRemoteDir.Length() > 0 &&
-		fCurrentRemoteDir[fCurrentRemoteDir.Length() - 1] == '/') {
-
-		url << "ftp://" << fHost << ":" << fPort
-			<< fCurrentRemoteDir << intName;
-	} else {
-		url << "ftp://" << fHost << ":" << fPort
-			<< fCurrentRemoteDir << "/" << intName;
+	url << "ftp://" << fHost << ":" << fPort << fCurrentRemoteDir;
+	if (fCurrentRemoteDir.Length() == 0
```

Comment:
Length() returns an int32, so it should be <= 0

Reply:
I don't think it would ever return a negative number, because it seems stupid. A string's length can't be negative.
Even though int32 CAN be a negative num. I don't know the implementation.

Reply:
well if you're at inverting a check, better do it right, so that reviewers don't need to ask themselves. Defensive programming.

Reply:
Got it. Will do that in the future.